src/adapters/registerChallengeAdapter.ts

```diff
@@ -0,0 +1,152 @@
+// enums
+import { Gender } from '../components/types/Profile';
+import { NewsletterType } from '../components/types/Newsletter';
+import { PaymentSubject } from '../components/enums/Payment';
+import { OrganizationType } from '../components/types/Organization';
+
+// types
+import type {
+  RegisterChallengeResult,
+  RegisterChallengePostPayload,
+} from '../components/types/ApiRegistration';
+import type { RegisterChallengePersonalDetailsForm } from '../components/types/RegisterChallenge';
+import type { ValidatedCoupon } from '../components/types/Coupon';
+
+/**
+ * Adapter for converting between API and store registration data formats
+ */
+export const registerChallengeAdapter = {
+  /**
+   * Convert API registration data to store format
+   * @param {RegisterChallengeResult} apiData - Registration data from API
+   * @returns {object} Store-compatible registration data
+   */
+  toStoreData(apiData: RegisterChallengeResult) {
+    const personalDetails: RegisterChallengePersonalDetailsForm = {
+      firstName: apiData.personal_details.first_name,
+      lastName: apiData.personal_details.last_name,
+      nickname: apiData.personal_details.nickname,
+      gender: apiData.personal_details.sex as Gender,
+      newsletter: this.parseNewsletterValues(
+        apiData.personal_details.newsletter,
+      ),
+      terms: apiData.personal_details.personal_data_opt_in,
+    };
+
+    return {
+      personalDetails,
+      organizationId: apiData.organization_id,
+      paymentSubject: apiData.personal_details
+        .payment_subject as PaymentSubject,
+      paymentAmount: apiData.personal_details.payment_amount
+        ? parseInt(apiData.personal_details.payment_amount)
```

Comment:
`payment_amount` field type should be integer type it was REST API bug.

Reply:
Updated type to `number`.